src/wikibase.py

```diff
@@ -267,58 +272,123 @@ def get_concept(
+        MAX_PAGE_REQUESTS = 2000  # Suitable up to 1M pages (500*2000)
 
-        params = {
+        base_params = {
             "action": "query",
             "format": "json",
-            "list": "allpages",  # See https://www.mediawiki.org/wiki/API:Allpages
+            "list": "allpages",
             "apnamespace": 120,
             "aplimit": PAGE_REQUEST_SIZE,
-            "apfilterredir": "nonredirects",  # Only fetch non-redirect pages
         }
+        # Update with any additional params supplied by the user
+        base_params.update(extra_params)
 
-        wikibase_ids = []
+        pages = []
         for i in range(MAX_PAGE_REQUESTS):
-            # Get and process a page into wikibase_ids
             response = self.session.get(
                 url=self.api_url,
-                params=params,
+                params=base_params,
             ).json()
-            wikibase_ids.extend(
-                [
-                    page["title"].replace("Item:", "")
-                    for page in response["query"]["allpages"]
-                ]
-            )
 
-            #  Handle errors / warnings, see:
-            # https://www.mediawiki.org/wiki/API:Continue#Example_3:_Python_code_for_iterating_through_all_results
             if "error" in response:
                 raise HTTPError(response["error"])
             if "warnings" in response:
                 logger.warning(response["warnings"])
 
-            # Handle pagination if there are more pages
+            batch_pages = response["query"]["allpages"]
+            pages.extend(batch_pages)
+
             if continue_params := response.get("continue"):
-                params.update(continue_params)
-                logger.info(
-                    f"Retrieved {len(wikibase_ids)} ids after page iteration {i}"
-                )
+                base_params.update(continue_params)
+                logger.info(f"Retrieved {len(pages)} pages after iteration {i}")
             else:
                 break
 
-        return wikibase_ids
+        return pages
+
+    def get_concept_ids(self) -> list[WikibaseID]:
+        """
+        Get concept ids from Wikibase.
+
+        :return list[WikibaseID]: The concept ids, e.g ["Q123", "Q456"]
+        """
+        pages = self._get_pages(extra_params={"apfilterredir": "nonredirects"})
+        return [page["title"].replace("Item:", "") for page in pages]
+
+    def get_all_redirects(self) -> dict[WikibaseID, WikibaseID]:
+        """
+        Get all redirects from Wikibase.
+
+        :return dict[WikibaseID, WikibaseID]: The redirects, e.g
+            {"Q123": "Q456", "Q456": "Q789"}
+        """
+        pages = self._get_pages(extra_params={"apfilterredir": "redirects"})
+        redirects = {}
+
+        # For each redirect, we need to find the wikibase ids of the source and target.
+        # We process the pages in batches of 50
+        for batch in [pages[i : i + 50] for i in range(0, len(pages), 50)]:
```

Comment:
What's the reason that this is 50? Would we ever want to change it (maybe via a function kwarg)?

Reply:
no specific reason tbh 🤷 it's a number, and it works! 

will add a kwarg with 50 as the default 🫡